platform/mbed_stats.h

```diff
@@ -36,6 +38,18 @@ typedef struct {
  */
 void mbed_stats_heap_get(mbed_stats_heap_t *stats);
 
+typedef struct {
+    uint32_t current_size;      /**< Current number of bytes used in all stacks. */
+    uint32_t max_size;          /**< Sum of the maximum number of bytes used in each stack. */
+    uint32_t reserved_size;     /**< Current number of bytes allocated for all stacks. */
+    uint32_t stack_cnt;         /**< Number of stacks currently allocated. */
+} mbed_stats_stack_t;
+
+/**
+ * Fill the passed in structure with stack stats.
+ */
+void mbed_stats_stack_get(mbed_stats_stack_t *stats);
```

Comment:
The use case is the same as for `mbed_stats_heap_get`: a general overview of where memory is being used on the system. This is useful for showing trends in the memory consumption of stack-allocated variables/classes.